storey/flow.py

```diff
@@ -1887,3 +1888,15 @@ async def _do(self, event):
             else:
                 event._metadata = metadata
             return await self._do_downstream(event)
+
+    def _verify_runnables(self, runnables: List[Union[str, ParallelExecutionRunnable]]):
+        """Verifies that the provided runnables are valid and registered."""
+        runnables_names = [r.name if isinstance(r, ParallelExecutionRunnable) else r for r in runnables]
+        register_runnables = [r.name if isinstance(r, ParallelExecutionRunnable) else r for r in self.runnables]
+        if not set(runnables_names).issubset(set(register_runnables)):
+            raise ValueError(
+                f"Runnables {set(runnables_names) - set(register_runnables)} are not part of the registered runnables"
+            )
```

Comment:
Suggested refactoring and error message improvement:
```suggestion
        runnable_names = {r.name if isinstance(r, ParallelExecutionRunnable) else r for r in runnables}
        registered_runnables = {r.name if isinstance(r, ParallelExecutionRunnable) else r for r in self.runnables}
        if not runnable_names.issubset(registered_runnables):
            unregistered_runnables_string = ",".join(runnable_names - registered_runnables)
            raise ValueError(f"The following selected Runnables are not registered: {unregistered_runnables_string}")
```

Reply:
We should also probably put `registered_runnables` in self, to avoid recalculating it. Number of runnables can be large, and this is run per event.

Reply:
@royischoss, my snippet above makes multiple changes, some of which you did not apply. Please update the code or let me know why you prefer not to make the suggested changes. 🙏

Reply:
I preferred caching in the init, and the unreachable code handling made me change the way I create the sets, because I am already iterating over the runnables